packages/protocol/test/signal/SignalService.integration.test.ts

```diff
@@ -0,0 +1,97 @@
+import { expect } from "chai";
+import { ethers } from "hardhat";
+import { TestHeaderSync } from "../../typechain";
+import { deploySignalService, getSignalProof } from "../utils/signal";
+import deployAddressManager from "../utils/addressManager";
+import { getBlockHeader } from "../utils/rpc";
+
+describe("integration:SignalService", function () {
+    async function deployIntegrationSignalServiceFixture() {
+        const [owner, nonOwner] = await ethers.getSigners();
+
+        const { chainId } = await ethers.provider.getNetwork();
+
+        const srcChainId = chainId;
+
+        // seondary node to deploy L2 on
+        const l2Provider = new ethers.providers.JsonRpcProvider(
+            "http://localhost:28545"
+        );
+
+        const l1Signer = await ethers.provider.getSigner();
+
+        const l2Signer = await l2Provider.getSigner(
+            (
+                await l2Provider.listAccounts()
+            )[0]
+        );
+
+        const l2NonOwner = await l2Provider.getSigner();
+
+        const l2Network = await l2Provider.getNetwork();
+        const enabledDestChainId = l2Network.chainId;
+
+        const addressManager = await deployAddressManager(l1Signer);
+
+        const l2AddressManager = await deployAddressManager(l2Signer);
+
+        const { signalService: l1SignalService } = await deploySignalService(
+            owner,
+            addressManager,
+            srcChainId
+        );
+
+        const { signalService: l2SignalService } = await deploySignalService(
+            l2Signer,
+            l2AddressManager,
+            enabledDestChainId
+        );
+
+        await l2AddressManager
+            .connect(l2Signer)
+            .setAddress(
+                `${srcChainId}.signal_service`,
+                l1SignalService.address
+            );
+
+        await addressManager
+            .connect(l2Signer)
+            .setAddress(
+                `${enabledDestChainId}.signal_service`,
+                l2SignalService.address
+            );
+
+        const headerSync: TestHeaderSync = await (
+            await ethers.getContractFactory("TestHeaderSync")
+        )
+            .connect(l2Signer)
+            .deploy();
+
+        await l2AddressManager
+            .connect(l2Signer)
+            .setAddress(`${enabledDestChainId}.taiko`, headerSync.address);
+
+        return {
+            l2Provider,
+            owner,
+            l1Signer,
+            l2Signer,
+            nonOwner,
+            l2NonOwner,
+            l1SignalService,
+            l2SignalService,
+            addressManager,
+            l2AddressManager,
+            enabledDestChainId,
+            srcChainId,
+            headerSync,
+        };
+    }
+
+    it("test", async function () {
+        const { l2Provider } = await deployIntegrationSignalServiceFixture();
```

Comment:
This is not a correct usage of fixture. You should use: `await loadFixture(deployIntegrationSignalServiceFixture)`, please see https://hardhat.org/hardhat-runner/docs/guides/test-contracts

Reply:
@dantaik I think in ./test/test_integration.sh we switched L1 to be an eth node and L2 to be the hardhat node, so loadFixture (which is a hardhat network helper) throws an error for me since it's being called on the eth network. Not sure how to go about fixing it.

Reply:
Then please do not call the function a fixture.